apps/desktop-wallet/src/components/Inputs/SelectOptionToken.tsx

```diff
@@ -31,25 +31,22 @@ type SelectOptionTokenProps = TokenDisplay & {
   className?: string
 }
 
-const SelectOptionToken = ({ isSelected, className, ...token }: SelectOptionTokenProps) => {
-  const { image, name, symbol, amount, decimals } = getTokenDisplayData(token)
+const SelectOptionToken = ({ isSelected, className, id, ...tokenProps }: SelectOptionTokenProps) => {
+  const { data: token } = useToken(id)
+
+  const name = isFT(token) || isNFT(token) ? token.name : undefined
+  const symbol = isFT(token) ? token.symbol : undefined
+  const amount = tokenProps.type !== 'NFT' ? tokenProps.availableBalance : undefined
```

Comment:
It's a funny mix between using type guards and using the `.type` property :)

Reply:
Yeah, it's until I fully migrate to one of the 2, it's temp